File: 3/3.py
def primeFactors(n):
  
  factors= list(filter(lambda x: (n % x == 0), [ x for x in range(2,n-1)])) 
  
  if len(factors) > 0:
    factors = [factors[0]] + primeFactors(int(n / factors[0]))
  else:
    return [n]
  
  return factors

print('35 prime factors', primeFactors(35)) 

File: 3/3.js
const primeFactors = (n) => {
  
    let factors = Array.from([...Array(n-1).keys()], x => x + 2).filter(x => n % x === 0)
    factors.pop() 
    
    if (factors.length > 0){
      factors = [factors[0]].concat(primeFactors(n / factors[0]))
    }else{
      return [n]
    }
  
    return factors
  }
  
  
  console.log('35', primeFactors(35))
  